shared/src/lib.rs

```diff
@@ -1,2 +1,3 @@
 pub mod arguments;
 pub mod tracing;
+pub mod valid_to;
```

Comment:
nanonit: `valid_to` is quite specific (the function can be used for more). Maybe `time` is a better name?